### PR TITLE
fix post api

### PR DESCRIPTION
**Pull request by ShadowJu**
pls!

### CODE REVIEW

File: routes/postRoutes.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-const postController = require('../controllers/postController');
+const postController = require("../controllers/postController");
 
-router.post('/createPost', postController.createPost);
+router.post("/createPost", postController.createPost);
```
Comment (osherlev) [inline routes/postRoutes.js:5]: revert it to ' '

File: controllers/postController.js [hunk 2]
```diff
@@ -1,71 +1,78 @@
...
-    } catch (error) {
-        res.status(500).json({error: error.message});
+  try {
+    const sender = req.query.sender;
+
+    if (!sender) {
+      return res.status(400).json({ message: "Sender parameter is required" });
     }
+    const posts = await Post.find({ sender });
+
```
Comment (osherlev) [inline controllers/postController.js:47]: add the deleted 
 if (posts.length === 0) {
            return res.status(404).json({message: 'No posts found for this sender'});
}

File: controllers/postController.js [hunk 3]
```diff
@@ -1,71 +1,78 @@
...
+    res.status(200).json(posts);
+  } catch (err) {
+    console.error(err);
```
Comment (osherlev) [inline controllers/postController.js:50]: remove

File: controllers/postController.js [hunk 4]
```diff
@@ -1,71 +1,78 @@
...
+    res.status(500).json({ error: "An error occurred while fetching posts" });
```
Comment (osherlev) [inline controllers/postController.js:51]: add error.message

File: controllers/postController.js [hunk 5]
```diff
@@ -1,71 +1,78 @@
...
-            return res.status(404).json({message: 'Post not found'});
-        }
-        res.status(200).json(post);
-    } catch (error) {
-        res.status(500).json({error: err.message});
-    }
+  try {
+    const { message, sender } = req.body;
+    const post = await Post.findByIdAndUpdate(
+      req.params.id,
```
Comment (osherlev) [inline controllers/postController.js:59]: I'm not sure why the 3 lines for parameters?

**Review by osherlev — COMMENTED** [review]
good for noticing!!!

File: controllers/postController.js [hunk 6]
```diff
@@ -1,71 +1,78 @@
...
 const getAllPosts = async (req, res) => {
-    const filter = req.query.owner;
-    let posts;
-    try {
-        posts = filter ? await Post.find({owner: filter}) : await Post.find();
-        res.send(posts);
-    } catch (error) {
-        res.status(400).send(error.message);
-    }
+  const filter = req.query.owner;
```
Comment (osherlev) [inline controllers/postController.js:14]: req.query.sender;

File: controllers/postController.js [hunk 7]
```diff
@@ -1,71 +1,78 @@
...
+  let posts;
+  try {
+    posts = filter ? await Post.find({ owner: filter }) : await Post.find();
```
Comment (osherlev) [inline controllers/postController.js:17]: sender

File: controllers/postController.js [hunk 8]
```diff
@@ -1,71 +1,78 @@
...
+      req.params.id,
+      { message, sender },
+      { new: true }
+    );
+    if (!post) {
+      return res.status(404).json({ message: "Post not found" });
+    }
+    res.status(200).json(post);
+  } catch (error) {
+    res.status(500).json({ error: err.message });
```
Comment (osherlev) [inline controllers/postController.js:68]: `error.message`

File: controllers/postController.js (hunk 5 above)
Comment (ShadowJu) [reply controllers/postController.js:59]: prettier extension 🤷‍♂️

File: routes/postRoutes.js (hunk 1 above)
Comment (ShadowJu) [reply routes/postRoutes.js:5]: prettier extension 🤷‍♂️

**Pull request merged**